File: 2023-2024/sourcecode/mod3/Lab7/ViewModelBasicsJava/app/src/main/java/com/example/viewmodelbasicsjava/ui/fragment/ListFragment.java
package com.example.viewmodelbasicsjava.ui.fragment;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.viewmodelbasicsjava.R;
import com.example.viewmodelbasicsjava.databinding.FragmentListBinding;

public class ListFragment extends Fragment {

    private FragmentListBinding binding;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentListBinding.inflate(inflater);



        return binding.getRoot();
    }
}

File: 2023-2024/sourcecode/mod4/lab13/Paging3BasicsKotlin/app/src/main/java/com/example/paging3basicskotlin/ui/ListFragment.kt
package com.example.paging3basicskotlin.ui

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.viewModels
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.paging3basicskotlin.databinding.FragmentListBinding
import com.example.paging3basicskotlin.viewmodel.SwapiViewModel
import kotlinx.coroutines.flow.collectLatest
import kotlinx.coroutines.launch

class ListFragment : Fragment() {

    private lateinit var binding: FragmentListBinding

    private val viewModel: SwapiViewModel by viewModels()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentListBinding.inflate(inflater)

        val characterAdapter = CharacterAdapter(CharacterComparator())

        viewLifecycleOwner.lifecycleScope.launch {
            viewModel.getData().collectLatest { pagingData ->
                characterAdapter.submitData(pagingData)

            }
        }

        binding.recycler.apply {
            layoutManager = LinearLayoutManager(requireContext())
            adapter = characterAdapter
            setHasFixedSize(true)
        }


        return binding.root
    }
}